src/Files.App/Data/Models/StatusBanner.cs

```diff
@@ -0,0 +1,196 @@
+		public ReturnResult Status
+		{
+			get => status;
+			set => SetProperty(ref status, value);
+		}
+
+		public FileOperationType Operation { get; private set; }
+
+		public string Message { get; private set; }
+
+		public InfoBarSeverity InfoBarSeverity { get; private set; } = InfoBarSeverity.Informational;
+
+		public string PrimaryButtonText { get; set; }
+
+		public string SecondaryButtonText { get; set; } = "Cancel";
+
+		public Action PrimaryButtonClick { get; }
+
+		public ICommand CancelCommand { get; }
+
+		public bool SolutionButtonsVisible { get; } = false;
+
+		public bool CancelButtonVisible
+			=> CancellationTokenSource is not null;
+
+		public CancellationTokenSource CancellationTokenSource { get; set; }
+
+		public string FullTitle
+		{
+			get => fullTitle;
+			set => SetProperty(ref fullTitle, value ?? string.Empty);
+		}
+
+		public bool IsCancelled
+		{
+			get => isCancelled;
+			set => SetProperty(ref isCancelled, value);
+		}
+
+		public StatusBanner(string message, string title, float progress, ReturnResult status, FileOperationType operation)
+		{
+			Message = message;
+			Title = title;
+			FullTitle = title;
+			initialProgress = progress;
+			Status = status;
+			Operation = operation;
+
+			CancelCommand = new RelayCommand(CancelOperation);
+
+			switch (Status)
+			{
+				case ReturnResult.InProgress:
+					IsProgressing = true;
+					if (string.IsNullOrWhiteSpace(Title))
+					{
+						switch (Operation)
+						{
+							case FileOperationType.Extract:
+								Title = "ExtractInProgress/Title".GetLocalizedResource();
+								break;
+
+							case FileOperationType.Copy:
+								Title = "CopyInProgress/Title".GetLocalizedResource();
+								break;
+
+							case FileOperationType.Move:
+								Title = "MoveInProgress".GetLocalizedResource();
+								break;
+
+							case FileOperationType.Delete:
+								Title = "DeleteInProgress/Title".GetLocalizedResource();
+								break;
+
+							case FileOperationType.Recycle:
+								Title = "RecycleInProgress/Title".GetLocalizedResource();
+								break;
+
+							case FileOperationType.Prepare:
+								Title = "PrepareInProgress".GetLocalizedResource();
+								break;
+						}
+					}
+
+					FullTitle = $"{Title} ({initialProgress}%)";
+
+					break;
+
+				case ReturnResult.Success:
+					IsProgressing = false;
+					if (string.IsNullOrWhiteSpace(Title) || string.IsNullOrWhiteSpace(Message))
+					{
+						throw new NotImplementedException();
+					}
+					else
+					{
+						FullTitle = Title;
+						InfoBarSeverity = InfoBarSeverity.Success;
+					}
+					break;
+
+				case ReturnResult.Failed:
+				case ReturnResult.Cancelled:
+					IsProgressing = false;
+					if (string.IsNullOrWhiteSpace(Title) || string.IsNullOrWhiteSpace(Message))
+					{
+						throw new NotImplementedException();
+					}
+					else
+					{
+						// Expanded banner
+						FullTitle = Title;
+						InfoBarSeverity = InfoBarSeverity.Error;
+					}
```

Comment:
We should merge these three cases and then set `InfoBarSeverity = Status is Success ? Success : Error`

Reply:
Fixed